src/BenchmarksApps/Websocket/Startup.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using System.Net.WebSockets;
+using System.Threading.Tasks;
+using System.Threading;
+using System;
+using System.Net;
+using System.Text;
+
+namespace BenchmarkServer
+{
+    public class Startup
+    {
+        private readonly IConfiguration _config;
+
+        public Startup(IConfiguration configuration)
+        {
+            _config = configuration;
+        }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddRouting();
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseRouting();
+            app.UseWebSockets();
+            app.Use(async (context, next) =>
+                {
+                    if (context.Request.Path == "/Echo")
+                    {
+                        if (context.WebSockets.IsWebSocketRequest)
+                        {
+                            using (WebSocket webSocket = await context.WebSockets.AcceptWebSocketAsync())
+                            {
+                                await Echo(context, webSocket);
+                            }
+                        }
+                        else
+                        {
+                            context.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                        }
+                    }
+                    else
+                    {
+                        await next();
+                    }
+                });
+        }
+
+        private async Task Echo(HttpContext context, WebSocket webSocket)
+        {
+            var receivedMessage = "";
+            var buffer = new byte[1024 * 4];
+            var result = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
+            while (true)
+            {
+                if (result.CloseStatus.HasValue)
+                {
+                    break;
+                }
+
+                receivedMessage += Encoding.UTF8.GetString(buffer, 0, result.Count);
+                if (result.EndOfMessage)
+                {
+                    await webSocket.SendAsync(Encoding.UTF8.GetBytes(receivedMessage).AsMemory(), result.MessageType, result.EndOfMessage, CancellationToken.None);
```

Comment:
Why create a string then convert it back to a byte array instead of just echoing the byte array?

Reply:
I don't mind changing it to just echoing byte array (actually, this was my initial code 😅). However, to clarify, my idea is that server behavior should be closer to some real-world server. I believe in most cases this would mean that messages have to be read up to an `EndOfMessage` flag before processing and returning a response. Another consideration is that this just an initial implementation that will be extended to other scenarios (e.g. compression), where I think this code will be useful (otherwise, it'll be re-implemented). Let me know what you think 🙂

Reply:
I believe the goal of the benchmark is to measure WebSockets, so we should try to do as little as possible outside of the WebSocket code so we know we are only measuring WebSockets.

When compression is added later there shouldn't need to be any change to the echo code.

In the future if we want to add more scenarios we can always update the code.